packages/sanity/src/_internal/cli/actions/schema/utils/schemaStoreValidation.ts

```diff
@@ -0,0 +1,235 @@
+import {type CliOutputter} from '@sanity/cli'
+import uniqBy from 'lodash/uniqBy'
+
+import {isDefined} from '../../../../manifest/manifestTypeHelpers'
+import {SANITY_WORKSPACE_SCHEMA_TYPE} from '../../../../manifest/manifestTypes'
+import {type DeleteSchemaFlags} from '../deleteSchemaAction'
+import {type SchemaListFlags} from '../listSchemasAction'
+import {type StoreSchemasFlags} from '../storeSchemasAction'
+import {getManifestDirPath} from './manifestReader'
+import {getStringArrayOutString} from './storeSchemaOutStrings'
+
+export const validForIdChars = 'a-zA-Z0-9._-'
+export const validForIdPattern = new RegExp(`^[${validForIdChars}]+$`, 'g')
+
+const requiredInId = SANITY_WORKSPACE_SCHEMA_TYPE.replace(/[.]/g, '\\.')
+const idPattern = new RegExp(
```

Comment:
if I'm reading this correctly, this would allow id's to both start with dashes, and also have consecutive dots, which is not supported: https://www.sanity.io/docs/ids (haven't checked if this is considered later on though)

Reply:
Added some additional checks: didn't touch the regex as not to complicate it further.